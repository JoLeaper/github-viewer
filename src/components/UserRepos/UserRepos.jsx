import React from 'react';
import PropTypes from 'prop-types';

const UserRepos = ({ userReposArray }) => (
  <>
    <h1>List of Repos</h1>
    <ul>
      {userReposArray.map(repo => {
        return <li key={repo}><a href={repo}>{repo}</a></li>;
      })}
    </ul>
  </>
);

UserRepos.propTypes = {
  userReposArray: PropTypes.array
};
export default UserRepos;
