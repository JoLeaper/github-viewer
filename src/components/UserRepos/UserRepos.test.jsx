import React from 'react';
import { shallow } from 'enzyme';
import UserRepos from './UserRepos';
const testArray = ['link1', 'link2', 'link3'];

describe('UserRepos component', () => {
  it('renders UserRepos', () => {
    const wrapper = shallow(<UserRepos userReposArray={testArray}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
  
