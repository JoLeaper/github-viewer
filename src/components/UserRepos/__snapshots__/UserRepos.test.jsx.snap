// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserRepos component renders UserRepos 1`] = `
<Fragment>
  <h1>
    List of Repos
  </h1>
  <ul>
    <li
      key="link1"
    >
      <a
        href="link1"
      >
        link1
      </a>
    </li>
    <li
      key="link2"
    >
      <a
        href="link2"
      >
        link2
      </a>
    </li>
    <li
      key="link3"
    >
      <a
        href="link3"
      >
        link3
      </a>
    </li>
  </ul>
</Fragment>
`;
